### DISCUSSION TITLE
difference between Batter-Toolkit, bclm and bclm_loop?

### DISCUSSION THREAD

**Discussion by ssh352** [mhaeuser/Battery-Toolkit#38] [Q&A] [answered]
https://github.com/zackelia/bclm
https://github.com/lslqtz/bclm_loop/tree/main

it is quite helpful to know the difference, thanks!

**Comment by mhaeuser** [comment, marked as answer]
Well, I don’t know these tools. From the looks of it, they try to use firmware-based limits, which used to be inflexible and seemingly broke entirely with Sequoia? I didn’t mess much with that, but if the dust settles around this, I might try implementing it as well. The loop version uses a literal background loop to continuously poll - not so nice. Battery Toolkit uses the OS APIs to be notified of changes to the charging level and is idling most of the time. Also Battery Toolkit can disable the power adapter and charge full on demand from the GUI menu.

**Reply by ssh352** [reply]
thanks!  it is very helpful
> Battery Toolkit uses the OS APIs to be notified of changes to the charging level and is idling most of the time. Also Battery Toolkit can disable the power adapter and charge full on demand from the GUI menu.

does it mean that Battery ToolKit will monitor battery level, if it drops from 80% for example a bit, it micro charges and keep it at 80%?

**Reply by mhaeuser** [reply]
It does whatever you configure it to do, you can check the settings in the README.

**Comment by piotrekdb-dev** [comment]
I got bclm on Macbook m1 and the question is. Do I have installed on Mac bclm and bclm loop and it start with system. Can i uninstall bclm because after update to sequoia I start use bclm loop
